executable-spec/src/Cardano/Ledger/Spec/STS/Update/Tallysip.hs

```diff
@@ -0,0 +1,222 @@
+                  then
+                    Data.VotingResult 0 0 0
+                  else
+                    Map.foldrWithKey'
+                      (\vkey conf Data.VotingResult { Data.stakeInFavor
+                                                    , Data.stakeAgainst
+                                                    , Data.stakeAbstain
+                                                    }
+                          -> let stake = stakeDist!vkey
+                             in case conf of
+                                 Data.For -> Data.VotingResult
+                                   (stakeInFavor + stake)
+                                   stakeAgainst
+                                   stakeAbstain
+                                 Data.Against -> Data.VotingResult
+                                   stakeInFavor
+                                   (stakeAgainst + stake)
+                                   stakeAbstain
+                                 Data.Abstain -> Data.VotingResult
+                                   stakeInFavor
+                                   stakeAgainst
+                                   (stakeAbstain + stake)
+                      )
+                      (Data.VotingResult 0 0 0)
+                      ballotsOfSIP
+          in vresips ⨃ [(sipHash, vResult)]
+
+        -- if approved then, update state of approved SIPs
+        apprvsips' =
+          if Data.stakeInFavor (vresips'!sipHash) > fromIntegral vThreshold
+            then
+              Set.insert sipHash apprvsips
+            else
+              apprvsips
+
+        -- if no-majority result, or no-quorum, calculate end of new voting period
+        -- and update active sips state
+        asips' =
+          if Data.stakeAbstain (vresips'!sipHash) > fromIntegral vThreshold
+             ||
+             (  Data.stakeInFavor (vresips'!sipHash) <= fromIntegral vThreshold
+             && Data.stakeAgainst (vresips'!sipHash) <= fromIntegral vThreshold
+             && Data.stakeAbstain (vresips'!sipHash) <= fromIntegral vThreshold
+             )
+            then -- a revoting is due - calc new voting period end
+              asips ⨃ [(sipHash, currentSlot `addSlot` (votPeriodEnd sipHash))]
+            else
+              asips
+        votPeriodEnd siphash =  Data.vpDurationToSlotCnt
+                                $ Data.votPeriodDuration
+                                . Data.metadata
+                                . Data.sipPayload
+                                $ (sipdb!siphash)
+
+      pure $ St { vresips = vresips'
+                , apprvsips = apprvsips'
+                , asips = asips'
+                }
+
+    ]
+
+-- | STS for tallying the votes of a
+-- bunch of SIPs
+data TALLYSIPS hashAlgo
+
+-- | Candidate SIPs for tallying
+-- data ToTally hashAlgo = ToTally ![(Data.SIPHash hashAlgo)]  -- ![Signal (TALLYSIP hashAlgo)]
+--   deriving (Eq, Show)
+
+type ToTally hashAlgo =  [Signal (TALLYSIP hashAlgo)]
```

Comment:
Do we need the type alias?

Reply:
It is a matter of style. What is better to say
`type Comment = Text`, or to use directly `Text`?